server-spi/src/main/java/org/keycloak/models/AbstractKeycloakTransaction.java

```diff
@@ -82,6 +82,7 @@ public enum TransactionState {
         NOT_STARTED, STARTED, ROLLBACK_ONLY, FINISHED
     }
 
+    protected void beginImpl() {}
```

Comment:
If breaking change is okay, then this method can be abstract as well.

Reply:
We shouldn't have breaking changes here, so it shouldn't be abstract.